pkg/agent/bakerapi_test.go

```diff
@@ -523,4 +526,33 @@ var _ = Describe("AgentBaker API implementation tests", func() {
 			}
 		})
 	})
+
+	Context("GetCachedVersionsOnVHD", func() {
+		It("should return cached VHD data", func() {
+			agentBaker, err := NewAgentBaker()
+			Expect(err).NotTo(HaveOccurred())
+
+			fromManifest, fromComponentsContainerImages, fromComponentsDownloadFiles, err := agentBaker.GetCachedVersionsOnVHD()
+			Expect(err).NotTo(HaveOccurred())
+
+			manifest := datamodel.CacheManifest()
+			component := datamodel.CacheComponents()
+
+			// The indices are hardcoded based on the current components.json.
```

Comment:
would we ever need to update these?

Reply:
Only if someone adds a new binary to components.json and messes up the indexing. Shouldn't matter if they add to the end of the file

Reply:
hmm, might want to add a note somewhere about that